### ISSUE TITLE
Embedding a note into another note not working

### ISSUE THREAD

**Issue by ChildishhAlbino** [HiDeoo/starlight-obsidian#7]
### Describe the bug

Title is pretty much it but when embedding one note into another like so
```md
![[Note]]
```
The generated page just has zero content where the embed is.

### To Reproduce

1. Create two pages
2. Embed one page inside another
3. Run the development server and review results

### Expected behavior

Per the documentation website; this is a supported feature so I would expect a similar outcome to what is shown on [this Obsidian wiki page.](https://help.obsidian.md/Linking+notes+and+files/Embed+files#Embed%20an%20audio%20file%20in%20a%20note)

### How often does this bug happen?

Every time

**Comment by HiDeoo** [comment]
Thanks for the report.

Would you be able to share a vault that reproduces the issue? It only needs to contain the minimum amount of data to reproduce the issue (including the `.obsidian` folder).

The test fixtures contain embedded notes for all types of link format for both Markdown links and Wiki links ([wikilink-shortest](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-wikilink-shortest/root%20embeds.md?plain=1#L13-L17) - [wikilink-relative](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-wikilink-relative/root%20embeds.md?plain=1#L13-L17) - [wikilink-absolute](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-wikilink-absolute/root%20embeds.md?plain=1#L13-L17) - [markdown-shortest](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-markdown-shortest/root%20embeds.md?plain=1#L13-L17) - [markdown-relative](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-markdown-relative/root%20embeds.md?plain=1#L13-L17) - [markdown-absolute](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-markdown-absolute/root%20embeds.md?plain=1#L13-L17)) and they seem to work as expected.

In fact, cloning the repository, installing the dependencies and running the `pnpm dev` command in the `example/` directory and opening the `http://localhost:4321/notes/root-embeds/` page displays embedded notes from various directories:

<img width="758" alt="image" src="https://github.com/HiDeoo/starlight-obsidian/assets/494699/54e20b55-2952-44f7-92db-e09bdc1893c2">

**Comment by ChildishhAlbino** [comment]
Interesting - I've tried removing everything from my Astro config that could be conflicting with no luck.

I believe I've managed to replicate the issue within the example project - how is the best way to give you that reproduction? Creating a separate branch in this repo or making an entirely different repo with the content as is?

**Comment by HiDeoo** [comment]
> how is the best way to give you that reproduction? Creating a separate branch in this repo or making an entirely different repo with the content as is?

I would say whatever is easier for your, even just a small repo containing the few files needed would be enough.

> However, I am confident I've understood the bug. It seems that for any file that is not at the root of the vault, the entire file path from the root of the vault has to be added for the embed to work.

Yeah, that's where the repro would come in handy, as for all the fixtures examples I shared in my previous post, there are actually duplicated 3 times: 1 time in a file at the root, 1 time in a file in a subfolder, and 1 time in a file in a subfolder of the subfolder, e.g. this [one](https://github.com/HiDeoo/starlight-obsidian/blob/main/fixtures/links-wikilink-shortest/folder/embeds%20in%20folder.md?plain=1#L13) for `wikilink-shortest` in a folder:

```
![[duplicate file name]]

![[folder/duplicate file name|duplicate file name]]

![[folder/nested folder/duplicate file name|duplicate file name]]
```

**Comment by ChildishhAlbino** [comment]
hey @HiDeoo - yep, I took a look at those different cases and that's how I came to my conclusion.

Happy to make a reproduction repo - I'll link it here sometime later this week.

**Comment by esmevane** [comment]
Just in the interests of supporting your notion, @ChildishhAlbino, I can confirm that I was able to reproduce this in a vault just today. I had the following structure:

```
books/
  book-one/
    index.md
    Introduction.md
```

And in `books/book-one/index.md`:

```md
## Introduction

![[Introduction]]
```

That didn't work. However, when I changed it to:

```md
## Introduction

![[books/book-one/Introduction]]
```

Then everything built and it referenced the embed correctly.

**Comment by ChildishhAlbino** [comment]
Thanks @esmevane - I haven't had the time to produce a minimal reproduction so glad to hear I'm not the only one having this issue - if you haven't already made a reproduction repo I'll try to endeavour to get that done this weekend - just depends what pops up.

**Comment by HiDeoo** [comment]
Thanks for the reproduction, I finally managed to track down the issue. This should be fixed in version [`0.4.2`](https://github.com/HiDeoo/starlight-obsidian/releases/tag/v0.4.2) of the plugin that I just released.
